source/osh/ui/view/View.js

```diff
@@ -131,6 +124,24 @@ export class View {
         observer.observe(this.elementDiv, {attributes: true});
     }
 
+    registerCallback() {
+        EventManager.observe(this.getEventName(), (event) => {
+            if (event.reset) {
+                this.reset(); // on data stream reset
+            } else {
+                this.setData(this.dataSourceId, event.data);
+            }
+        });
+    }
+
+    unregisterCallback() {
+        EventManager.remove(this.getEventName());
```

Comment:
I think this would unregister all views that are listening to the same event, not only this one.
We need to somehow send a reference to the view or to the callback itself to EventMap so only this one can be removed.

Reply:
You are absolutely right; I see 2 ways to do that:
1) Register with an extra view-id: EventManager.observe(this.getEventName(), cb, viewId)
or
2) Register with specific name designing the view id without changing the current way: EventManager.observe(this.getEventName()+'-'+viewId, cb)

Reply:
I've added a specific id in some case to specify the case where we have to find the callback corresponding to a view. Put 'any' by default